mlflow/pyfunc/__init__.py

```diff
@@ -2453,6 +2442,13 @@ def _predict_row_batch(predict_fn, args):
             result_values = _convert_array_values(result, result_type)
             return pandas.Series(result_values)
 
+        if isinstance(result_type, SparkStructType):
+            if isinstance(result, list) and isinstance(result[0], pydantic.BaseModel):
```

Comment:
q: is it possible that an empty list comes to this branch?

Reply:
Emm good question 🤔 normally no, but let me add this condition